helix-term/src/commands/lsp.rs

```diff
@@ -1324,3 +1331,191 @@ pub fn select_references_to_symbol_under_cursor(cx: &mut Context) {
+            view.last_inlay_hint_request_id = None;
+            for doc in editor.documents.values_mut() {
+                doc.reset_all_text_annotations();
+            }
+        }
+        return;
+    }
+
+    for (view, _) in editor.tree.views_mut() {
+        let doc = match editor.documents.get_mut(&view.doc) {
+            Some(doc) => doc,
+            None => continue,
+        };
+        if let Some(callback) = compute_inlay_hints_for_view(view, doc) {
+            jobs.callback(callback);
+        }
+    }
+}
+
+fn compute_inlay_hints_for_view(
+    view: &mut View,
+    doc: &mut Document,
+) -> Option<std::pin::Pin<Box<impl Future<Output = Result<crate::job::Callback, anyhow::Error>>>>> {
+    let view_id = view.id;
+    let doc_id = view.doc;
+    let doc_revision = doc.get_current_revision();
+
+    let language_server = doc.language_server()?;
+
+    let capabilities = language_server.capabilities();
+
+    let (future, new_inlay_hint_request) = match capabilities.inlay_hint_provider {
+        Some(
+            lsp::OneOf::Left(true)
+            | lsp::OneOf::Right(lsp::InlayHintServerCapabilities::Options(_)),
+        ) => {
+            let doc_text = &doc.text();
+            let len_lines = doc_text.len_lines();
+
+            // Compute ~3 times the current view height of inlay hints, that way some scrolling
+            // will not show half the view with hints and half without while still being faster
+            // than computing all the hints for the full file (which could be dozens of time
+            // longer than the view is).
+            let view_height = view.inner_height();
+            let first_visible_line = view.estimate_last_doc_line(doc).saturating_sub(view_height);
+            let first_line = first_visible_line
+                .saturating_sub(view_height)
+                .min(len_lines);
+            let last_line = first_visible_line
+                .saturating_add(view_height)
+                .min(len_lines);
+
+            let new_inlay_hint_request = LastInlayHintRequestId {
+                doc_id,
+                doc_revision,
+                first_line,
+                last_line,
+            };
+            // Don't recompute the annotations in case nothing has changed about the view
+            if view.last_inlay_hint_request_id == Some(new_inlay_hint_request) {
+                return None;
+            }
+
+            let doc_slice = doc_text.slice(..);
+            let first_char_in_range = doc_slice.line_to_char(first_line);
+            let last_char_in_range = doc_slice.line_to_char(last_line);
+
+            let range = helix_lsp::util::range_to_lsp_range(
+                doc_text,
+                helix_core::Range::new(first_char_in_range, last_char_in_range),
+                language_server.offset_encoding(),
+            );
+
+            (
+                language_server.text_document_range_inlay_hints(doc.identifier(), range, None),
```

Comment:
I think it would be a bit cleaner if we passed a `helix_core::Range` to `text_document_range_inlay_hints` and do the lsp conversion inside `text_document_range_inlay_hints` in `helix-lsp`

Reply:
Eh, I don't like that: for one, it's not the style used by the other functions in this file, and it conflates `lsp::Range` and `helix_core::Range` in place where only LSP-related stuff happens, I don't think that's good for clarity compared to the single conversion call later in the crate chain